proxy/InkAPI.cc

```diff
@@ -8256,6 +8265,17 @@ TSHttpTxnConfigStringSet(TSHttpTxn txnp, TSOverridableConfigKey conf, const char
     if (value && length > 0) {
       s->t_state.txn_conf->client_cert_filepath = const_cast<char *>(value);
     }
+  case TS_CONFIG_HTTP_RESPONSE_VIA_TRANSPORT:
+    data._s.setView(value, length);
+    if (ViaTransportVerbosityConversion(s->t_state.txn_conf->response_via_transport, "proxy.config.http.response_via_tranport",
+                                        RECD_STRING, data)) {
+    }
+    break;
+  case TS_CONFIG_HTTP_REQUEST_VIA_TRANSPORT:
+    data._s.setView(value, length);
+    if (ViaTransportVerbosityConversion(s->t_state.txn_conf->request_via_transport, "proxy.config.http.request_via_tranport",
```

Comment:
I would expect an enum RecType would find rather wide spread use. There are at least three places now where it would be used and I would expect more in the future if that style were made easy. I also expect to move in that direction as we change over to Lua in any case, using Lua symbolic constants. The other benefit would be it would make possible to specify by tag *or* by numeric value. Right now that's difficult to do, it should be easy. In addition, it would make it easy to address zwoop's performance concern.